plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/bedrock-runtime.ts

```diff
@@ -98,32 +145,61 @@ export class BedrockRuntimeServiceExtension implements ServiceExtension {
     response: NormalizedResponse,
     span: Span,
     tracer: Tracer,
-    config: AwsSdkInstrumentationConfig
+    config: AwsSdkInstrumentationConfig,
+    startTime: HrTime
   ) {
     if (!span.isRecording()) {
       return;
     }
 
     switch (response.request.commandName) {
       case 'Converse':
-        return this.responseHookConverse(response, span, tracer, config);
+        return this.responseHookConverse(
+          response,
+          span,
+          tracer,
+          config,
+          startTime
+        );
     }
   }
 
   private responseHookConverse(
     response: NormalizedResponse,
     span: Span,
     tracer: Tracer,
-    config: AwsSdkInstrumentationConfig
+    config: AwsSdkInstrumentationConfig,
+    startTime: HrTime
   ) {
     const { stopReason, usage } = response.data;
+
+    const sharedMetricAttrs: Attributes = {
+      [ATTR_GEN_AI_SYSTEM]: GEN_AI_SYSTEM_VALUE_AWS_BEDROCK,
+      [ATTR_GEN_AI_OPERATION_NAME]: GEN_AI_OPERATION_NAME_VALUE_CHAT,
+      [ATTR_GEN_AI_REQUEST_MODEL]: response.request.commandInput.modelId,
+    };
+
+    const durationSecs =
+      hrTimeToMilliseconds(hrTimeDuration(startTime, hrTime())) / 1000;
+    this.operationDuration.record(durationSecs, sharedMetricAttrs);
+
     if (usage) {
       const { inputTokens, outputTokens } = usage;
       if (inputTokens !== undefined) {
         span.setAttribute(ATTR_GEN_AI_USAGE_INPUT_TOKENS, inputTokens);
+
+        this.tokenUsage.record(inputTokens, {
+          ...sharedMetricAttrs,
+          [ATTR_GEN_AI_TOKEN_TYPE]: GEN_AI_TOKEN_TYPE_VALUE_INPUT,
+        });
       }
       if (outputTokens !== undefined) {
         span.setAttribute(ATTR_GEN_AI_USAGE_OUTPUT_TOKENS, outputTokens);
+
+        this.tokenUsage.record(outputTokens, {
+          ...sharedMetricAttrs,
+          [ATTR_GEN_AI_TOKEN_TYPE]: GEN_AI_TOKEN_TYPE_VALUE_COMPLETION,
```

Comment:
`export const GEN_AI_TOKEN_TYPE_VALUE_COMPLETION = "output"` is weird: "COMPLETION" rather than "OUTPUT". 
Ah, looks like that'll be fixed by https://github.com/open-telemetry/semantic-conventions/pull/1757/files#diff-0936f2641601c5b370b10dfb33c8f82d6d731a2663cf042eef2aa5114c720902R177
which is probably in semconv v1.31.0, for which the OTel JS semconv package doesn't have a release yet.
So eventually this'll be `GEN_AI_TOKEN_TYPE_VALUE_OUTPUT`.

Reply:
There *is* a semconv v1.31.0 release now, so you *could*:

1. update package.json to bump the dep to `"@opentelemetry/semantic-conventions": "^1.31.0"`
2. change the usage and import in bedrock-runtime.ts to `GEN_AI_TOKEN_TYPE_VALUE_OUTPUT`
3. regenerate the src/semconv.ts by re-running (I hope this works, it is meant to):

```
cd .../plugins/node/opentelemetry-instrumentation-aws-sdk
~/src/opentelemetry-js-contrib/scripts/gen-semconv-ts.js
```